File: ort-frontend/src/types/question.ts
export interface Question {
  id: number;
  title: string;
  text: string;
  access_code: string;
  is_closed: boolean;
  created_at: string;
  close_date?: string | null;
  answer_count: number;
  student_id?: string;
  answer?: string | null;
}

export interface QuestionCreate {
  title: string;
  text: string;
  access_code: string;
}

export interface QuestionResponse {
  id: number;
  title: string;
  text: string;
  access_code: string;
  is_closed: boolean;
  close_date?: string | null;
  message?: string;
}

export interface QuestionAccess {
  student_id: string;
}

export interface QuestionDetails {
  id: number;
  title: string;
  text: string;
  access_code: string;
  status: 'Open' | 'Closed';
}

export interface QuestionWithAnswers {
  question: Question;
  answers: import('./answer').Answer[];
  answer_count: number;
}

File: backend/app/database/models/question.py
"""
Question database model.
SQLAlchemy ORM model for question entities in the classroom Q&A application.
"""

from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime
from datetime import datetime
from typing import Optional
from .base import Base
from ...utils.timezone import now_israel


class Question(Base):
    """
    SQLAlchemy model for questions table.
    
    Represents a question in the classroom Q&A system.
    """
    
    __tablename__ = "questions"
    
    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    title = Column(Text, nullable=False, index=True)
    text = Column(Text, nullable=False)
    access_code = Column(String, nullable=False, unique=True, index=True)
    is_closed = Column(Integer, nullable=False, default=0)  # 0 = False/Open, 1 = True/Closed
    created_at = Column(DateTime, nullable=False, default=now_israel)
    close_date = Column(DateTime, nullable=True, default=None)
    
    def __repr__(self):
        return f"<Question(id={self.id}, title='{self.title}', access_code='{self.access_code}', is_closed={self.is_closed}, close_date={self.close_date})>"
